pkg/controllers/namespace/namespace_sync_controller_test.go

```diff
@@ -0,0 +1,505 @@
+		name      string
+		namespace string
+		skipped   []*regexp.Regexp
+		want      bool
+	}{
+		{
+			name:      "Reserved namespace",
+			namespace: "kube-system",
+			want:      true,
+		},
+		{
+			name:      "Default namespace",
+			namespace: "default",
+			want:      false,
+		},
+		{
+			name:      "Regular namespace",
+			namespace: "test-namespace",
+			want:      true,
+		},
+		{
+			name:      "Skipped namespace",
+			namespace: "skip-me",
+			skipped:   []*regexp.Regexp{regexp.MustCompile("skip-.*")},
+			want:      false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c := &Controller{
+				SkippedPropagatingNamespaces: tt.skipped,
+			}
+			got := c.namespaceShouldBeSynced(tt.namespace)
+			assert.Equal(t, tt.want, got)
+		})
+	}
+}
+
+func TestController_Reconcile(t *testing.T) {
+	scheme := runtime.NewScheme()
+	_ = corev1.AddToScheme(scheme)
+	_ = clusterv1alpha1.Install(scheme)
+	_ = policyv1alpha1.Install(scheme)
+	_ = workv1alpha1.Install(scheme)
+
+	tests := []struct {
+		name           string
+		namespace      *corev1.Namespace
+		clusters       []clusterv1alpha1.Cluster
+		expectedResult controllerruntime.Result
+		expectedError  bool
+	}{
+		{
+			name: "Namespace should be synced",
+			namespace: &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-namespace",
+				},
+			},
+			clusters: []clusterv1alpha1.Cluster{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "cluster1",
+					},
+				},
+			},
+			expectedResult: controllerruntime.Result{},
+			expectedError:  false,
+		},
+		{
+			name: "Namespace should not be synced",
+			namespace: &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "kube-system",
+				},
+			},
+			expectedResult: controllerruntime.Result{},
+			expectedError:  false,
+		},
+		{
+			name: "Namespace with skip auto propagation label",
+			namespace: &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "skip-namespace",
+					Labels: map[string]string{
+						policyv1alpha1.NamespaceSkipAutoPropagationLabel: "true",
+					},
+				},
+			},
+			expectedResult: controllerruntime.Result{},
+			expectedError:  false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fakeClient := fake.NewClientBuilder().WithScheme(scheme).WithObjects(tt.namespace).WithLists(&clusterv1alpha1.ClusterList{Items: tt.clusters}).Build()
+			fakeRecorder := record.NewFakeRecorder(100)
+
+			c := &Controller{
+				Client:          fakeClient,
+				EventRecorder:   fakeRecorder,
+				OverrideManager: overridemanager.New(fakeClient, fakeRecorder),
+			}
+
+			req := controllerruntime.Request{
+				NamespacedName: types.NamespacedName{
+					Name: tt.namespace.Name,
+				},
+			}
+
+			result, err := c.Reconcile(context.Background(), req)
+			assert.Equal(t, tt.expectedResult, result)
+			assert.Equal(t, tt.expectedError, err != nil)
+
+			if tt.name == "Namespace should be synced" {
```

Comment:
This judgment method is not good. The case name may change. Can you use another method?

Reply:
Okay, onto it.